sw/host/opentitantool/src/command/certificate.rs

```diff
@@ -80,36 +81,73 @@ impl CommandDispatch for CodegenCommand {
         _context: &dyn Any,
         _transport: &TransportWrapper,
     ) -> Result<Option<Box<dyn Annotate>>> {
-        let template = load_template(&self.template)?;
-        // Generate C and header files.
-        let (output_c, output_h) = if let Some(output_dir) = &self.output_dir {
-            (
-                output_dir.join(format!("{}.c", &template.name)),
-                output_dir.join(format!("{}.h", &template.name)),
-            )
-        } else {
-            (
-                self.output_c
-                    .clone()
-                    .expect("--output-c must be specified when --output-dir is not used"),
-                self.output_h
-                    .clone()
-                    .expect("--output-h must be specified when --output-dir is not used"),
-            )
-        };
-        let mut output_c = File::create(output_c)?;
-        let mut output_h = File::create(output_h)?;
+        match self.cert_format {
+            CertFormat::X509 => {
+                let template = load_template(&self.template)?;
+                // Generate C and header files.
+                let (output_c, output_h) = if let Some(output_dir) = &self.output_dir {
```

Comment:
This can be factored between the two cases

Reply:
by "factored" do you mean shared? if so I agree :)